Queue tweaks

gfr10598 opened the pull request:
Add more batch queues.
Add retry params for batch queues to improve behavior.  Backoff causes sidestream to pulse badly.
Increase workers to 15, which is a good compromise between SS and NDT.

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/etl/485)
<!-- Reviewable:end -->

stephen-soltesz commented:
Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---

*[appengine/queue.yaml, line 89 at r2](https://beta.reviewable.io/reviews/m-lab/etl/485#-L965XNd7VKvpNuMtc5r:-L965XNd7VKvpNuMtc5s:b-nq0uwq) ([raw file](https://github.com/m-lab/etl/blob/cb1786cf7a1750fe223b122caa271a6df8c2f6e1/appengine/queue.yaml#L89)):*
> ```YAML
> #    This should allow 30 days of data to be processed within one day, with
> #    some headroom for future growth.
> # TODO - misnomer - we are currently using these queues for sidestream as well.
> ```

Is the plan or batch queues to be shared? -- basically, what is the TODO?

---

*[appengine/queue.yaml, line 96 at r2](https://beta.reviewable.io/reviews/m-lab/etl/485#-L966mFb6NrBsVTy5xxL:-L966mFb6NrBsVTy5xxM:b-bbyoas) ([raw file](https://github.com/m-lab/etl/blob/cb1786cf7a1750fe223b122caa271a6df8c2f6e1/appengine/queue.yaml#L96)):*
> ```YAML
>   max_concurrent_requests: 40
>   retry_parameters:
>     task_age_limit: 8h
> ```

Is 8h adequate? The text above aims for 1 data-month / calendar-day. I understand the increased number of queues to mean that on average each queue will have 2 data-days worth of tasks. 

I understood from the docs that the timer does not start until the first failure. And, that the backoff seconds should retry quickly. But, we've observed that task queues do not order task runs.  does max backoff guarantee an earlier retry? Or merely cap the backoff delay? Are we guaranteed that 8h is adequate to guarantee a retry before timeout?

---

*[appengine/queue.yaml, line 99 at r2](https://beta.reviewable.io/reviews/m-lab/etl/485#-L966PiFBNLfOH2CsXXj:-L966PiFBNLfOH2CsXXk:butzp1t) ([raw file](https://github.com/m-lab/etl/blob/cb1786cf7a1750fe223b122caa271a6df8c2f6e1/appengine/queue.yaml#L99)):*
> ```YAML
>     min_backoff_seconds: 30
>     max_backoff_seconds: 120
>     max_doublings: 0
> ```

if `max_doublings` is zero can the backoff period ever grow beyond 30s?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/m-lab/etl/485)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/m-lab/etl/485#-:-L968BGr3aaM71rSmwAW:bnfp4nl)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[appengine/queue.yaml, line 89 at r2](https://beta.reviewable.io/reviews/m-lab/etl/485#-L965XNd7VKvpNuMtc5r:-L96d5kSAR6M8pf3L_Gy:b6qdnov) ([raw file](https://github.com/m-lab/etl/blob/cb1786cf7a1750fe223b122caa271a6df8c2f6e1/appengine/queue.yaml#L89)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

Is the plan or batch queues to be shared? -- basically, what is the TODO?
</blockquote></details>

Added clarification.

---

*[appengine/queue.yaml, line 96 at r2](https://beta.reviewable.io/reviews/m-lab/etl/485#-L966mFb6NrBsVTy5xxL:-L96czDiEX751wHMO_Gq:b-ce0jye) ([raw file](https://github.com/m-lab/etl/blob/cb1786cf7a1750fe223b122caa271a6df8c2f6e1/appengine/queue.yaml#L96)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

Is 8h adequate? The text above aims for 1 data-month / calendar-day. I understand the increased number of queues to mean that on average each queue will have 2 data-days worth of tasks. 

I understood from the docs that the timer does not start until the first failure. And, that the backoff seconds should retry quickly. But, we've observed that task queues do not order task runs.  does max backoff guarantee an earlier retry? Or merely cap the backoff delay? Are we guaranteed that 8h is adequate to guarantee a retry before timeout?
</blockquote></details>

Added more comments to clarify, and increased to 12 hours, which I think is very conservative.

---

*[appengine/queue.yaml, line 99 at r2](https://beta.reviewable.io/reviews/m-lab/etl/485#-L966PiFBNLfOH2CsXXj:-L96DmZ13fr5Eerg2aUG:bzhhq5p) ([raw file](https://github.com/m-lab/etl/blob/cb1786cf7a1750fe223b122caa271a6df8c2f6e1/appengine/queue.yaml#L99)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

if `max_doublings` is zero can the backoff period ever grow beyond 30s?
</blockquote></details>

It grows linearly after it reaches max_doublings.  I'd rather not document that here, thinking it is better for people to go look at the task queue docs.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/m-lab/etl/485)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[appengine/queue.yaml, line 99 at r2](https://beta.reviewable.io/reviews/m-lab/etl/485#-L966PiFBNLfOH2CsXXj:-L96gTIi5OxNOEHQp3qY:b-bizqh8) ([raw file](https://github.com/m-lab/etl/blob/cb1786cf7a1750fe223b122caa271a6df8c2f6e1/appengine/queue.yaml#L99)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

It grows linearly after it reaches max_doublings.  I'd rather not document that here, thinking it is better for people to go look at the task queue docs.
</blockquote></details>

Added comment.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/m-lab/etl/485)*
<!-- Sent from Reviewable.io -->
